runhouse/rns/secrets.py

```diff
@@ -562,19 +672,32 @@ def read_secrets(cls, from_env: bool = False, file_path: Optional[str] = None):
                 f"Reading secrets from env is not supported for {cls.PROVIDER_NAME}"
             )
         else:
-            creds_file = file_path or cls.CREDENTIALS_FILE
+            creds_file = file_path or cls.default_credentials_path()
             config_data = cls.read_yaml_file(creds_file)
             token = config_data["github.com"]["oauth_token"]
 
         return {"provider": cls.PROVIDER_NAME, "token": token}
 
     @classmethod
-    def save_secrets(cls, secrets: dict, file_path: Optional[str] = None):
-        dest_path = file_path or cls.CREDENTIALS_FILE
+    def save_secrets(
+        cls, secrets: dict, file_path: Optional[str] = None, overwrite: bool = False
+    ):
+        dest_path = file_path or cls.default_credentials_path()
         config = cls.read_yaml_file(dest_path) if cls.file_exists(dest_path) else {}
         config["github.com"] = {"oauth_token": secrets["token"]}
 
-        cls.save_to_yaml_file(config, dest_path)
+        if overwrite:
+            cls.save_to_yaml_file(config, dest_path)
+            cls.save_secret_to_config()
+
+
+class Providers(Enum):
+    AWS = AWSSecrets()
+    AZURE = AzureSecrets()
+    GCP = GCPSecrets()
+    HUGGINGFACE = HuggingFaceSecrets()
+    LAMBDA = LambdaSecrets()
+    SKY = SkySecrets()
```

Comment:
Love it, nice solution. Maybe better to move it to the top of the file above the Secrets class, that's the convention I'm used to seeing for Enums.